compiler/x/i386/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3620,11 +3620,13 @@ TR::Register *OMR::X86::I386::TreeEvaluator::lternaryEvaluator(TR::Node *node, T
    TR::Register *falseReg = cg->evaluate(falseVal);
    TR::Register *trueReg  = cg->longClobberEvaluate(trueVal);
 
-   TR::ILOpCodes condOp = condition->getOpCodeValue();
-   if((condOp == TR::icmpeq) || (condOp == TR::icmpne))
+   auto condOp = condition->getOpCode();
+   bool longCompare = (condition->getOpCode().isBooleanCompare() &&
+         (condition->getFirstChild()->getOpCode().isLong() || condition->getSecondChild()->getOpCode().isLong()));
+   if(condOp.isCompareForEquality() && condition->getFirstChild()->getOpCode().isIntegerOrAddress() && !longCompare)
```

Comment:
Similarly, it may be worthy bring `!longCompare` as the first condition, so that the rest checks can be skipped when `!longCompare` is false.

Reply:
addressed.